tests/unit/preprocessor/test_preprocessor_file.py

```diff
@@ -0,0 +1,163 @@
+"""Unit tests for :class:`esmvalcore.preprocessor.PreprocessorFile`."""
+
+from unittest import mock
+
+import pytest
+from iris.cube import Cube, CubeList
+
+from esmvalcore.preprocessor import PreprocessorFile
+
+ATTRIBUTES = {
+    'filename': 'file.nc',
+    'standard_name': 'precipitation',
+    'long_name': 'Precipitation',
+    'short_name': 'pr',
+    'units': 'kg m-2 s-1',
+    'frequency': 'mon',
+}
+
+
+@pytest.fixture
+def product():
+    """PreprocessorFile object used for testing."""
+    cube = Cube(
+        0,
+        var_name='tas',
+        standard_name='air_temperature',
+        long_name='Near-Surface Air Temperature',
+        units='K',
+        attributes={'frequency': 'day'},
+    )
+    product = PreprocessorFile(attributes=ATTRIBUTES, settings={})
+    product._cubes = CubeList([cube, cube, cube])
+    return product
+
+
+def test_update_attributes_empty_cubes(product):
+    """Test ``update_attributes``."""
+    product._cubes = CubeList([])
+    product.update_attributes()
+
+    assert not product._cubes
+    assert product.attributes == ATTRIBUTES
+
+
+def test_update_attributes(product):
+    """Test ``update_attributes``."""
+    product.update_attributes()
+
+    assert product.attributes == {
+        'filename': 'file.nc',
+        'standard_name': 'air_temperature',
+        'long_name': 'Near-Surface Air Temperature',
+        'short_name': 'tas',
+        'units': 'K',
+        'frequency': 'day',
+    }
+    assert isinstance(product.attributes['units'], str)
+
+
+def test_update_attributes_empty_standard_name(product):
+    """Test ``update_attributes``."""
+    product._cubes[0].standard_name = None
+    product.update_attributes()
+
+    assert product.attributes == {
+        'filename': 'file.nc',
+        'standard_name': '',
+        'long_name': 'Near-Surface Air Temperature',
+        'short_name': 'tas',
+        'units': 'K',
+        'frequency': 'day',
+    }
+    assert isinstance(product.attributes['units'], str)
+
+
+def test_update_attributes_empty_long_name(product):
+    """Test ``update_attributes``."""
+    product._cubes[0].long_name = None
+    product.update_attributes()
+
+    assert product.attributes == {
+        'filename': 'file.nc',
+        'standard_name': 'air_temperature',
+        'long_name': '',
+        'short_name': 'tas',
+        'units': 'K',
+        'frequency': 'day',
+    }
+    assert isinstance(product.attributes['units'], str)
+
+
+def test_update_attributes_empty_var_name(product):
+    """Test ``update_attributes``."""
+    product._cubes[0].var_name = None
+    product.update_attributes()
+
+    assert product.attributes == {
+        'filename': 'file.nc',
+        'standard_name': 'air_temperature',
+        'long_name': 'Near-Surface Air Temperature',
+        'short_name': '',
+        'units': 'K',
+        'frequency': 'day',
+    }
+    assert isinstance(product.attributes['units'], str)
```

Comment:
This would be easier to maintain if it was parametrized using pytest

Reply:
So do we want to keep the current behavior regarding `None` names? Set them to `''`?

Reply:
There's probably some NCL diagnostics depending on that.. so I think yes?

Reply:
My question is rather: do we update the names at all if they are `None` or do we keep the original value in this case?

Reply:
I imagine this could be useful if a preprocessor modifies a variable in such a way that the `standard_name` is no longer valid? For the other attributes, I'm less sure if it is useful to have a mechanism to unset them.

Reply:
I also think having this in there is good, but we have to keep in mind that this might lead to unexpected changes (https://github.com/ESMValGroup/ESMValCore/issues/651).

Reply:
That looks like a bug in a preprocessor function, right? This change will mean that bug is going to propagate into the metadata.yml file, but if the bug was fixed things should work fine again.

Reply:
Yes, that's definitely a bug, just wanted to mention that this might cause problems when testing actual recipes.